pkg/commands/types/enums/enums.go

```diff
@@ -1,22 +1,106 @@
 package enums
 
-type RebaseMode int
+import "github.com/jesseduffield/lazygit/pkg/i18n"
 
+type WorkingTreeState int
+
+// Note: this is a bitfield, several of these can be set at once
 const (
-	// this means we're neither rebasing nor merging
-	REBASE_MODE_NONE RebaseMode = iota
-	// this means normal rebase as opposed to interactive rebase
-	REBASE_MODE_NORMAL
-	REBASE_MODE_INTERACTIVE
-	// REBASE_MODE_REBASING is a general state that captures both REBASE_MODE_NORMAL and REBASE_MODE_INTERACTIVE
-	REBASE_MODE_REBASING
-	REBASE_MODE_MERGING
+	// this means we're neither rebasing nor merging, cherry-picking, or reverting
+	WORKING_TREE_STATE_NONE     WorkingTreeState = 0
```

Comment:
I don't like how we're using the same type (`WorkingTreeState`) for the bits as for the bitfield. Naively I would expect that I could directly check `if workingTreeState == WORKING_TREE_STATE_CHERRY_PICKING` but that's really only going to return true if we're only cherry picking and not also merging. I also can't do `if workingTreeState == WORKING_TREE_STATE_CHERRY_PICKING || workingTreeState == WORKING_TREE_STATE_REBASING` to ask if it's either cherry picking or rebasing.

So I'm wondering, how about we just have a struct of booleans:
```go
type WorkingTreeState struct {
  rebasing bool
  merging bool
  cherryPicking bool
  reverting bool
}
```

And then define methods on that struct.

Reply:
Good point. Yes, a struct is an improvement to this.

I couldn't do this as a fixup, so I rebased and force-pushed. I added dceb1960f026 as a preparation, and then d2573dc5dc93 is the changed commit. The rest of the branch is unchanged except for adaptions to the new API.